src/client/Microsoft.Identity.Client/ManagedIdentity/AzureArcManagedIdentitySource.cs

```diff
@@ -0,0 +1,99 @@
+            string imdsEndpoint = EnvironmentVariables.ImdsEndpoint;
+
+            // if BOTH the env vars IDENTITY_ENDPOINT and IMDS_ENDPOINT are set the MsiType is Azure Arc
+            if (string.IsNullOrEmpty(identityEndpoint) || string.IsNullOrEmpty(imdsEndpoint))
+            {
+                return default;
+            }
+
+            if (!Uri.TryCreate(identityEndpoint, UriKind.Absolute, out Uri endpointUri))
+            {
+                throw new MsalClientException(MsalError.InvalidManagedIdentityEndpoint, MsalErrorMessage.ManagedIdentityEndpointInvalidUriError);
+            }
+
+            return new AzureArcManagedIdentitySource(endpointUri, requestContext);
+        }
+
+        private AzureArcManagedIdentitySource(Uri endpoint, RequestContext requestContext) : base(requestContext)
+        {
+            _endpoint = endpoint;
+            _clientId = requestContext.ServiceBundle.Config.ManagedIdentityUserAssignedClientId;
+            _resourceId = requestContext.ServiceBundle.Config.ManagedIdentityUserAssignedResourceId;
+
+            if (!string.IsNullOrEmpty(_clientId) || !string.IsNullOrEmpty(_resourceId))
+            {
+                throw new MsalClientException(MsalError.UserAssignedManagedIdentityNotSupported, MsalErrorMessage.ManagedIdentityUserAssignedNotSupported);
+            }
+        }
+
+        protected override ManagedIdentityRequest CreateRequest(string resource)
+        {
+            ManagedIdentityRequest request = new ManagedIdentityRequest(System.Net.Http.HttpMethod.Get, _endpoint);
+
+            request.Headers.Add("Metadata", "true");
+            request.QueryParameters["api-version"] = ArcApiVersion;
+            request.QueryParameters["resource"] = resource;
+
+            return request;
+        }
+
+        protected override async Task<ManagedIdentityResponse> HandleResponseAsync(
+            AppTokenProviderParameters parameters,
+            HttpResponse response,
+            CancellationToken cancellationToken)
+        {
+            if (response.StatusCode == System.Net.HttpStatusCode.Unauthorized)
+            {
+                if (!response.HeadersAsDictionary.TryGetValue("WWW-Authenticate", out string challenge))
+                {
+                    throw new MsalServiceException(MsalError.ManagedIdentityRequestFailed, MsalErrorMessage.ManagedIdentityNoChallengeError);
+                }
+
+                var splitChallenge = challenge.Split(new char[] { '=' }, StringSplitOptions.RemoveEmptyEntries);
+
+                if (splitChallenge.Length != 2)
+                {
+                    throw new MsalServiceException(MsalError.ManagedIdentityRequestFailed, MsalErrorMessage.ManagedIdentityInvalidChallange);
+                }
+
+                var authHeaderValue = "Basic " + File.ReadAllText(splitChallenge[1]);
+
+                ManagedIdentityRequest request = CreateRequest(ScopeHelper.ScopesToResource(parameters.Scopes.ToArray()));
```

Comment:
nit: Let's make sure that the new API `AcquireTokenForClientByManagedIdentity` (or whatever name we decide on) will have a single string scope API.